mpp/shmem.h.in

```diff
@@ -18,6 +18,10 @@
 
 #include <stddef.h>
 #include <complex.h>
```

Comment:
The C/C++ interop with complex is weird...

C11 officially defines a compiler keyword _Complex, and the complex.h header defines a macro #define complex _Complex
In g++ at least, complex.h doesn't do that when compiling with C++, and instead behaves like the C++11 <complex> header file, which defines a class "complex<T>" that name-clashes with the macro if the macro is created too early in the file. So by putting the #define (and the #undef at line 498) inside #ifdef __cplusplus, the C declarations still work.

C++ code using "float _Complex" will probably work (but might not because _Complex is technically C, not C++). "float complex" won't work because of the name clash. There's a stackoverflow post that claims C++'s std::complex and C's _Complex are supposed to have identical memory layouts though, so that might be a route for proper C++ complex bindings.